src/shared/props.ts

```diff
@@ -243,7 +247,46 @@ type PAriaAllAttributes = PAriaWidgetAttributes &
     PAriaDragAndDropAttributes &
     PAriaRelationshipAttributes;
 
+/**
+ * implements padding and margin spacing properties to a components properties
+ * definition, where they are needed.
+ *
+ * Is part of the type PGlobals, but can be used separately.
+ */
+type PGlobalsLayout = {
+    /**
+     * padding according to `TSpacingDefinition` type
+     */
+    padding?: TSpacingDefinition;
+    /**
+     * margin according to `TSpacingDefinition` type
+     */
+    margin?: TSpacingDefinition;
+};
+
+/**
+ * implements className and children properties to a components properties
+ * definition, where they are needed.
+ *
+ * Is part of the type PGlobals, but can be used separately.
+ */
+type PGlobalsMisc = {
+    /**
+     * custom className
+     */
+    className?: string;
+    /**
+     * component children
+     */
+    children?: ReactNode | ReactNodeArray;
```

Comment:
what about adding event handles and aria props here as well?

Reply:
Event handlers could be an option, but depending on the situation a handling function can have a totally different structure